src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs

```diff
@@ -233,6 +234,7 @@ public int MaxAutomaticRedirections
 		/// If <c>true</c> then the server requested authorization and the application must use information
 		/// found in <see cref="RequestedAuthentication"/> to set the value of <see cref="PreAuthenticationData"/>
 		/// </summary>
+		[MemberNotNullWhen(true, nameof(RequestedAuthentication))]
```

Comment:
This fails the Xamarin.Android build:
```
src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs(237,4): error CS0246: The type or namespace name 'MemberNotNullWhenAttribute' could not be found (are you missing a using directive or an assembly reference?) [src/Mono.Android/Mono.Android.csproj::TargetFramework=monoandroid10]
src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs(237,4): error CS0246: The type or namespace name 'MemberNotNullWhen' could not be found (are you missing a using directive or an assembly reference?) [src/Mono.Android/Mono.Android.csproj::TargetFramework=monoandroid10]
```

Can you change this to:
```suggestion
#if NETCOREAPP
		[MemberNotNullWhen(true, nameof(RequestedAuthentication))]
#endif
```

Thanks for your contribution!

Reply:
Thank you for spotting that! I should've known that monoandroid10 and net7.0 are still built from the same branch.